src/components/ProductGrid.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState } from 'react'
+import { colors } from '../styles/colors'
+import { spacing } from '../styles/spacing'
+import { typography } from '../styles/typography'
+import product from '@/data/product'
+
+const ProductGrid = () => {
+  const [showAll, setShowAll] = useState(false)
+  const ranks = showAll ? [...Array(21).keys()].map(i => i + 1) : [...Array(6).keys()].map(i => i + 1)
```

Comment:
```suggestion
const allRanks = [...Array(21).keys()].map(i => i + 1);
const commonRanks = [...Array(6).keys()].map(i => i + 1);
const ProductGrid = () => {
  const [showAll, setShowAll] = useState(false)
  const ranks = showAll ? allRanks : commonRanks;
```

이렇게 컴포넌트 밖에 리액트 의존성이 없는 부분은 분리해주시는게 좋답니다!
렌더링 관점에서, 이렇게 하면 어떤 점이 다른걸까요?

Reply:
emotion css속성을 사용하는것과 같은 효과가 있을 것 같습니다